src/base/misc/ioctl.c

```diff
@@ -148,16 +148,33 @@ void
 add_to_io_select_new(int new_fd, void (*func)(void *), void *arg,
 	const char *name)
 {
-    int flags;
+    int flags, ret;
+
     if ((new_fd+1) > numselectfd) numselectfd = new_fd+1;
     if (numselectfd > MAX_FD) {
 	error("Too many IO fds used.\n");
 	leavedos(76);
     }
     flags = fcntl(new_fd, F_GETFL);
-    fcntl(new_fd, F_SETOWN, getpid());
-    fcntl(new_fd, F_SETFL, flags | O_ASYNC);
-    fcntl(new_fd, F_SETFD, FD_CLOEXEC);
+    if (flags == -1) {
```

Comment:
You can just do
if (!fcntl() || !fcntl() || !fcntl()) error();
to noe write too much.

Reply:
okay

Reply:
Is this better?
```diff
-    flags = fcntl(new_fd, F_GETFL);
-    fcntl(new_fd, F_SETOWN, getpid());
-    fcntl(new_fd, F_SETFL, flags | O_ASYNC);
-    fcntl(new_fd, F_SETFD, FD_CLOEXEC);
+
+    if ((flags = fcntl(new_fd, F_GETFL)) == -1 ||
+            fcntl(new_fd, F_SETOWN, getpid()) == -1 ||
+            fcntl(new_fd, F_SETFL, flags | O_ASYNC) == -1 ||
+            fcntl(new_fd, F_SETFD, FD_CLOEXEC) == -1) {
+       error("add_to_io_select_new: Fcntl failed\n");
+       leavedos(76);
+    }
+
```